meeshkan/serve/mock/faker/stateless_faker.py

```diff
@@ -0,0 +1,305 @@
+import json
+import random
+import typing
+from functools import reduce
+from typing import Any, Mapping, Sequence, Union, cast
+
+from faker import Faker
+from http_types import Request, Response
+from openapi_typed_2 import OpenAPIObject, Reference, convert_from_openapi
+
+from meeshkan.serve.mock.faker.faker_base import FakerBase
+from meeshkan.serve.mock.faker.faker_exception import FakerException
+from meeshkan.serve.mock.matcher import (
+    change_ref,
+    change_refs,
+    get_response_from_ref,
+    ref_name,
+)
+from meeshkan.serve.mock.storage.mock_data import MockData
+from meeshkan.serve.utils.opanapi_utils import get_x
+
+
+class StatelessFaker(FakerBase):
```

Comment:
I share Mike's sentiment from the earlier review that this doesn't seem to me like something that should be a class? It doesn't contain state (I think because it even says so:)). It's not a big deal but I'm a bit confused already about what behaviour comes from where. And because one rule of Pragmatic Programming is "don't use inheritance unless needed", just wanted to raise the comment :)

Reply:
That's because we need to change the behaviour of a protected method (fake_ref) in a recursion chain preserving others as they are. That's exactly the case for inheritance.

Reply:
Hmm to be that rather sounds like one needs to inject `fake_ref` behaviour? But I don't understand the code well enough to comment more, I believe if you say one needs inheritance here :) Just for someone seeing the code for the first time, it's really hard to understand the `StatefulFaker -> StatelessFaker -> FakerBase` inheritance chain and understanding where all the different behaviours come from and if they're hard-coded somewhere higher up the chain or not.

Reply:
`FakerBase` is just an abstract interface to interact with the outside world in a type-safe manner.
And the only two things `StatefulFaker` does are overriding of `fake_ref`  and `_update_data`. Inheritance seems like a nice way to implement this injection. But we can do it any other way if there're better ways.